microbench/src/test/java/io/netty/microbench/http2/Http2FrameWriterBenchmark.java

```diff
@@ -0,0 +1,562 @@
+import io.netty.handler.codec.http2.DefaultHttp2ConnectionEncoder;
+import io.netty.handler.codec.http2.DefaultHttp2FrameReader;
+import io.netty.handler.codec.http2.DefaultHttp2FrameWriter;
+import io.netty.handler.codec.http2.Http2Connection;
+import io.netty.handler.codec.http2.Http2ConnectionHandler;
+import io.netty.handler.codec.http2.Http2FrameAdapter;
+import io.netty.handler.codec.http2.Http2FrameWriter;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.Attribute;
+import io.netty.util.AttributeKey;
+import io.netty.util.NetUtil;
+import io.netty.util.concurrent.DefaultEventExecutor;
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.Future;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.Random;
+
+import org.junit.AfterClass;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+
+@State(Scope.Benchmark)
+public class Http2FrameWriterBenchmark extends AbstractMicrobenchmark {
+    private static final EnvironmentParameters NIO_UNPOOLED_PARAMS = new EnvironmentParameters() {
+        @Override
+        public EventLoopGroup newEventLoopGroup() {
+            return new NioEventLoopGroup();
+        }
+
+        @Override
+        public Class<? extends ServerChannel> serverChannelClass() {
+            return NioServerSocketChannel.class;
+        }
+
+        @Override
+        public Class<? extends Channel> clientChannelClass() {
+            return NioSocketChannel.class;
+        }
+
+        @Override
+        public ByteBufAllocator clientAllocator() {
+            return UnpooledByteBufAllocator.DEFAULT;
+        }
+    };
+    private static final EnvironmentParameters NIO_POOLED_PARAMS = new EnvironmentParameters() {
+        @Override
+        public EventLoopGroup newEventLoopGroup() {
+            return new NioEventLoopGroup();
+        }
+
+        @Override
+        public Class<? extends ServerChannel> serverChannelClass() {
+            return NioServerSocketChannel.class;
+        }
+
+        @Override
+        public Class<? extends Channel> clientChannelClass() {
+            return NioSocketChannel.class;
+        }
+
+        @Override
+        public ByteBufAllocator clientAllocator() {
+            return PooledByteBufAllocator.DEFAULT;
+        }
+    };
+    private static final Environment[] environments = { boostrapEnv(NIO_POOLED_PARAMS),
+            boostrapEnv(NIO_UNPOOLED_PARAMS), boostrapEnvWithTransport(NIO_POOLED_PARAMS),
+            boostrapEnvWithTransport(NIO_UNPOOLED_PARAMS) };
+    private static final ByteBuf[] dataPayload = { createData(256), createData(DEFAULT_MAX_FRAME_SIZE),
+            createData(MAX_FRAME_SIZE_UPPER_BOUND), createData(10 * MAX_FRAME_SIZE_UPPER_BOUND) };
+
+    @Param({ "0", "1", "2", "3" })
+    public int dataPayloadIndex;
```

Comment:
from @nmittler (so I don't forget):

Use an enum so the results will be descriptive about what is being benchmarked.